doc/releases/migration-guide-3.6.rst

```diff
@@ -32,6 +32,10 @@ Bootloader
 Bluetooth
 =========
 
+  * Mesh: Deprecated :kconfig:option:`CONFIG_BT_MESH_PROV_DEVICE`. This option is
```

Comment:
Each change should be on a new line, example:
https://github.com/zephyrproject-rtos/zephyr/blob/bfff0f9f4cdf8f32454c87b0c15a16be4ca1fc4f/doc/releases/release-notes-3.5.rst?plain=1#L201-L204